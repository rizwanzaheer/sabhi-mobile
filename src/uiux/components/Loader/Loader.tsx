import * as React from 'react'
import { ActivityIndicator } from 'react-native'
import Container from '../Container/Container'
import Text from '../Text/Text'
import Device from '../../services/device'

interface LoaderProps {
  bottom?: number
  left?: number
  zIndex?: number
  text?: string
  width?: number
  size?: number | 'large' | 'small' | undefined
}

const Loader: React.FC<LoaderProps> = ({ bottom, left, zIndex, text, width, size }) => {
  const defaultWidth = 200

  return (
    <Container
      flexDirection={'row'}
      alignItems={'center'}
      padding={10}
      br={20}
      viewStyle={{
        position: 'absolute',
        bottom: bottom || 50,
        zIndex: zIndex || 500,
        left: left || Device.width / 2 - (width || defaultWidth) / 2,
        width: width || defaultWidth,
        backgroundColor: 'rgba(0,0,0,0.8)',
      }}
    >
      <Container marginRight={10}>
        <ActivityIndicator size={size} />
      </Container>
      {text && <Text textColor={'#ffffff'}>{text}</Text>}
    </Container>
  )
}

export default Loader
