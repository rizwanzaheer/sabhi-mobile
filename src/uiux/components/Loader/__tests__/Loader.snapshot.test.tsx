import * as React from 'react'
import { render } from 'react-native-testing-library'
import Loader from '../Loader'

describe('Component(snapshots): Loader', () => {
  it('should render with no props', () => {
    const tree = render(<Loader />)

    expect(tree).toMatchSnapshot()
  })

  it('should render with all props', () => {
    const tree = render(
      <Loader bottom={100} left={100} zIndex={100} text={'Loader test'} width={200} size={'large'} />,
    )

    expect(tree).toMatchSnapshot()
  })
})
