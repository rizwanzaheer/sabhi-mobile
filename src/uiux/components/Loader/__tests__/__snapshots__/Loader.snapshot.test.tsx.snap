// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component(snapshots): Loader should render with all props 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0.8)",
      "borderRadius": 20,
      "bottom": 100,
      "flex": undefined,
      "flexDirection": "row",
      "height": undefined,
      "justifyContent": undefined,
      "left": 100,
      "margin": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "padding": 10,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "position": "absolute",
      "width": 200,
      "zIndex": 100,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": 10,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "flexWrap": "wrap",
      }
    }
  >
    Loader test
  </Text>
</View>
`;

exports[`Component(snapshots): Loader should render with no props 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0.8)",
      "borderRadius": 20,
      "bottom": 50,
      "flex": undefined,
      "flexDirection": "row",
      "height": undefined,
      "justifyContent": undefined,
      "left": 275,
      "margin": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "padding": 10,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "position": "absolute",
      "width": 200,
      "zIndex": 500,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": 10,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
    />
  </View>
</View>
`;
